daemon/inertia/auth/password.go

```diff
@@ -0,0 +1,59 @@
+package auth
+
+import (
+	"errors"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+var (
+	errSameUsernamePassword = errors.New("Username and password must be different")
+	errInvalidUsername      = errors.New("Only letters, numbers and underscore are allowed in usernames")
+	errInvalidPassword      = errors.New("Only letters, numbers and underscore are allowed in passwords, and password must be at least 5 characters")
+)
+
+func hashPassword(password string) (string, error) {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return "", errors.New("bcrypt password hashing unsuccessful: " + err.Error())
+	}
+	return string(hash), nil
+}
+
+func correctPassword(hash string, password string) bool {
+	return bcrypt.CompareHashAndPassword([]byte(hash), []byte(password)) == nil
+}
+
+func validateCredentialValues(username, password string) error {
+	if username == password {
+		return errSameUsernamePassword
+	}
+	if len(password) < 5 {
+		return errInvalidPassword
+	}
+	validChars := "abcdefghijklmnopqrstuvwxyzæøåABCDEFGHIJKLMNOPQRSTUVWXYZÆØÅ_0123456789"
+
```

Comment:
what are all these extra characters (æøå)?

Reply:
Not sure, they were in a random "list of valid characters" I found - I get rid of the weirder ones